bi_sql_editor/models/bi_sql_view.py

```diff
@@ -286,32 +284,39 @@ def button_create_sql_view_and_model(self):
             sql_view._create_index()
 
             if sql_view.is_materialized:
-                sql_view.cron_id = (
-                    self.env["ir.cron"].create(sql_view._prepare_cron()).id
-                )
+                if not sql_view.cron_id:
+                    sql_view.cron_id = (
+                        self.env["ir.cron"].create(sql_view._prepare_cron()).id
+                    )
+                else:
+                    sql_view.cron_id.active = True
             sql_view.state = "model_valid"
 
     def button_set_draft(self):
-        for sql_view in self:
+        for sql_view in self.filtered(lambda x: x.state != "draft"):
             sql_view.menu_id.unlink()
             sql_view.action_id.unlink()
             sql_view.tree_view_id.unlink()
             sql_view.graph_view_id.unlink()
             sql_view.pivot_view_id.unlink()
             sql_view.search_view_id.unlink()
-            if sql_view.cron_id:
-                sql_view.cron_id.unlink()
 
             if sql_view.state in ("model_valid", "ui_valid"):
                 # Drop SQL View (and indexes by cascade)
                 if sql_view.is_materialized:
                     sql_view._drop_view()
 
+                if sql_view.cron_id:
+                    sql_view.cron_id.active = False
+
                 # Drop ORM
                 sql_view._drop_model_and_fields()
 
             sql_view.write({"state": "draft", "has_group_changed": False})
```

Comment:
you should remove this line, because calling super do it.
see : https://github.com/OCA/reporting-engine/pull/519/files#diff-355785372a18bc17c4d5d51dcc29c551f88bdd3e13b7911276b07c1cec5792f8L289